aqt/archives.py

```diff
@@ -407,12 +406,13 @@ def _parse_update_xml(self, os_target_folder, update_xml_text, target_packages:
         for packageupdate in package_updates:
             if not self.all_extra:
                 target_packages.remove_module_for_package(packageupdate.name)
-            should_filter_archives: bool = self.subarchives and self.should_filter_archives(packageupdate.name)
+            should_filter_archives: bool = (self.subarchives is not None) and self.should_filter_archives(packageupdate.name)
 
             for archive in packageupdate.downloadable_archives:
                 archive_name = archive.split("-", maxsplit=1)[0]
-                if should_filter_archives and archive_name not in self.subarchives:
-                    continue
+                if self.subarchives is not None:
+                    if should_filter_archives and archive_name not in self.subarchives:
+                        continue
```

Comment:
```suggestion
                if should_filter_archives and bool(self.subarchives) and archive_name not in self.subarchives:
                    continue
```

Reply:
I reverted this change because it caused this error:
```
aqt/archives.py:415:74: error: Unsupported right operand type for in ("Optional[Iterable[str]]")  [operator]
                    if should_filter_archives and bool(self.subarchives) and archive_name not in self.subarchives:
                                                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```